File: VB/DevExpress.AI.Samples.Blazor/Data/IssuesContext.vb
Imports Microsoft.EntityFrameworkCore

Namespace DevExpress.AI.Samples.Blazor.Data

    Public Class IssuesContext
        Inherits DbContext

        Public Sub New(ByVal options As DbContextOptions(Of IssuesContext))
            MyBase.New(options)
        End Sub

        Public Property Items As DbSet(Of Issue)

        Public Property Projects As DbSet(Of Project)

        Public Property Users As DbSet(Of User)

        Protected Overrides Sub OnModelCreating(ByVal modelBuilder As ModelBuilder)
            modelBuilder.Entity(Of Issue)(Function(entity)
                entity.HasKey(Function(t) New With {t.ID})
                ' Properties
                entity.Property(Function(t) t.ID)
                entity.Property(Function(t) t.Name).HasMaxLength(50)
                entity.Property(Function(t) t.Description).HasMaxLength(2147483647)
                entity.Property(Function(t) t.Resolution).HasMaxLength(2147483647)
            End Function)
            modelBuilder.Entity(Of Project)(Function(entity)
                entity.HasKey(Function(t) t.ID)
                ' Properties
                entity.Property(Function(t) t.ID)
                entity.Property(Function(t) t.Name).HasMaxLength(100)
            End Function)
            modelBuilder.Entity(Of User)(Function(entity)
                entity.HasKey(Function(t) t.ID)
                ' Properties
                entity.Property(Function(t) t.ID)
                entity.Property(Function(t) t.FirstName).HasMaxLength(25)
                entity.Property(Function(t) t.LastName).HasMaxLength(25)
                entity.Property(Function(t) t.Country).HasMaxLength(15)
                entity.Property(Function(t) t.City).HasMaxLength(15)
                entity.Property(Function(t) t.Address).HasMaxLength(60)
                entity.Property(Function(t) t.Phone).HasMaxLength(24)
                entity.Property(Function(t) t.Email).HasMaxLength(50)
            End Function)
        End Sub
    End Class
End Namespace

File: CS/DevExpress.AI.Samples.Blazor/Data/IssuesContext.cs
using Microsoft.EntityFrameworkCore;

namespace DevExpress.AI.Samples.Blazor.Data {
    public class IssuesContext : DbContext {
        public IssuesContext(DbContextOptions<IssuesContext> options)
            : base(options) {
        }

        public DbSet<Issue> Items { get; set; }
        public DbSet<Project> Projects { get; set; }
        public DbSet<User> Users { get; set; }

        protected override void OnModelCreating(ModelBuilder modelBuilder) {
            modelBuilder.Entity<Issue>(entity => {
                entity.HasKey(t => new { t.ID });
                // Properties
                entity.Property(t => t.ID);
                entity.Property(t => t.Name)
                    .HasMaxLength(50);
                entity.Property(t => t.Description)
                    .HasMaxLength(2147483647);
                entity.Property(t => t.Resolution)
                    .HasMaxLength(2147483647);
            });
            modelBuilder.Entity<Project>(entity => {
                entity.HasKey(t => t.ID);
                // Properties
                entity.Property(t => t.ID);
                entity.Property(t => t.Name)
                    .HasMaxLength(100);
            });
            modelBuilder.Entity<User>(entity => {
                entity.HasKey(t => t.ID);
                // Properties
                entity.Property(t => t.ID);
                entity.Property(t => t.FirstName)
                    .HasMaxLength(25);
                entity.Property(t => t.LastName)
                    .HasMaxLength(25);
                entity.Property(t => t.Country)
                    .HasMaxLength(15);
                entity.Property(t => t.City)
                    .HasMaxLength(15);
                entity.Property(t => t.Address)
                    .HasMaxLength(60);
                entity.Property(t => t.Phone)
                    .HasMaxLength(24);
                entity.Property(t => t.Email)
                    .HasMaxLength(50);
            });
        }
    }
}
